vault/resource_mount.go

```diff
@@ -236,8 +305,32 @@ func updateMount(d *schema.ResourceData, meta interface{}, excludeType bool) err
 		path = newPath
 	}
 
-	if d.HasChange("allowed_managed_keys") {
-		config.AllowedManagedKeys = expandStringSlice(d.Get("allowed_managed_keys").(*schema.Set).List())
+	if d.HasChange(consts.FieldAllowedManagedKeys) {
```

Comment:
Out of curiosity, is there a reason we are conditionally setting these new parameters, rather than always setting them?

Reply:
It's mostly because they all have varied minimum version requirements. We can't always set them since some fields are only enabled on certain Vault versions.

The idea is that all these params will only be added in if the user provides them / makes a change to them. The expectation is that if a user is providing us a field to use, then they are aware of their Vault Server and its API features/limitations. We can also individually add version checks for all the older versions to the parameters,I figured this would be easier and less "If this version but less than that version do this, else that"